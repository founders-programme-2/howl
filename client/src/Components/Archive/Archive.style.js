import styled from 'styled-components';

export const BouncerContainer = styled.div`
  margin: 10% auto;
`;

export const ScrollButton = styled.button`
  font-weight: bold;
  position: fixed;
  font-size: 20px;
  top: 15%;
  right: 2%;
  font-family: 'Raleway', sans-serif;
  cursor: pointer;
`;
