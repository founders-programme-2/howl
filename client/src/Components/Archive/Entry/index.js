import React from 'react';
import {
  Article,
  HeaderFour,
  Para,
  DetailsHeader,
  DetailsPara,
} from './Entry.style';

const Entry = ({ id, title, year, category, details, tags, viewFullPost }) => {
  const shortenDetails = storyDetails => {
    if (storyDetails) {
      const shortenP = storyDetails.split(' ').slice(0, 29);
      let shortenedBody = shortenP.join(' ');
      shortenedBody += '...';
      return shortenedBody;
    }
    return null;
  };

  const renderTags = tags
    ? tags.map(tag => <Para key={tag}>{tag}</Para>)
    : null;

  return (
    <Article onClick={viewFullPost(id)}>
      <h3>{title || 'Untitled'}</h3>
      <div aria-label="[Date of Story]">
        <HeaderFour>Date of story:</HeaderFour>
        <Para>{year}</Para>
      </div>
      <div aria-label="Category">
        <HeaderFour>Category:</HeaderFour>
        <Para>{category}</Para>
      </div>
      <div aria-label="[Tags]">
        <HeaderFour>Tags:</HeaderFour>
        {renderTags}
      </div>
      <div aria-label="[Details]">
        <DetailsHeader>Details:</DetailsHeader>
        <DetailsPara>{shortenDetails(details)}</DetailsPara>
      </div>
    </Article>
  );
};

export default Entry;
