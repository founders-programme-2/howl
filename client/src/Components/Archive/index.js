import React, { Fragment, Component } from 'react';
import axios from 'axios';
import { Link } from 'react-router-dom';
import Footer from '../Common/Footer';
import navigationUrls from '../../constants/navigationUrls';
import Entry from './Entry/index';
import BounceLoaderComponent from '../BounceLoader';
import BouncerContainer from './Archive.style';

class Archive extends Component {
  constructor(props) {
    super(props);
  }

  state = {
    results: [],
    selectedPostId: '',
    loadingFlag: false,
  };

  componentDidMount() {
    this.setState({ loadingFlag: true }, () => {
      axios.get('/archive/feed').then(response => {
        this.setState({ results: response.data.data, loadingFlag: false });
      });
    });
  }

  viewFullPost = id => () => {
    this.setState({ selectedPostId: id }, () => {
      const { history } = this.props;
      const { selectedPostId } = this.state;
      history.push(`/story/${selectedPostId}`);
    });
  };

  render() {
    const { TIMELINE_URL } = navigationUrls;
    const { results, loadingFlag } = this.state;
    const { setFilters } = this.props;
    const renderResultsAsEntries = results
      ? results.map(result => (
          <Entry
            key={result.id}
            id={result.id}
            title={result.fields.title}
            year={result.fields.year}
            category={result.fields.category}
            details={result.fields.details}
            tags={result.fields.tags}
            viewFullPost={this.viewFullPost}
          />
        ))
      : null;

    const EntriesOrLoader = loadingFlag ? (
      <BouncerContainer>
        <BounceLoaderComponent loadingFlag={loadingFlag} />
      </BouncerContainer>
    ) : (
      <Fragment>{renderResultsAsEntries}</Fragment>
    );
    return (
      <Fragment>
        <main>
          <header>
            <h1>The HOWL Archive</h1>
          </header>
          <p>
            Browse our story and photo archive by using the archive controls at
            the bottom of this page to filter your results by category,
            location, tags or year.
          </p>
          <p>
            Prefer a visual approach? Don&#39;t worry our{' '}
            <Link to={TIMELINE_URL}>timeline</Link> is coming soon!
          </p>
          <h2>Recent contributions</h2>
          {EntriesOrLoader}
        </main>
        <Footer setFilters={setFilters} />
      </Fragment>
    );
  }
}

export default Archive;
