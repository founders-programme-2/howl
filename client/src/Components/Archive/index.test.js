import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Archive from './index';

test('Natural', () => {
  const archive = renderer
    .create(
      <BrowserRouter>
        {' '}
        <Archive />{' '}
      </BrowserRouter>
    )
    .toJSON();
  expect(archive).toMatchSnapshot();
});
