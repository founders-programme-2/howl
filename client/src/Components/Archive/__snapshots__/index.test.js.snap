// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Natural 1`] = `
Array [
  " ",
  <main>
    <header>
      <h1>
        The HOWL Archive
      </h1>
    </header>
    <p>
      Browse our story and photo archive by using the archive controls at the bottom of this page to filter your results by category, location, tags or year.
    </p>
    <p>
      Prefer a visual approach? Don't worry our
       
      <a
        href="/timeline"
        onClick={[Function]}
      >
        timeline
      </a>
       is coming soon!
    </p>
    <h2>
      Recent contributions
    </h2>
    <div
      className="sc-fjdhpX dvNwif"
    >
      <div
        className="sweet-loading"
      >
        <div
          className="css-1evy0nx-wrapper-wrapper"
        >
          <div
            className="css-z7rh7d-style"
          />
          <div
            className="css-jib95i-style"
          />
        </div>
      </div>
    </div>
  </main>,
  <button
    className="sc-jzJRlG dnqxAy"
    onClick={[Function]}
  >
     
    Jump To Search
     
  </button>,
  <footer
    className="sc-bdVaJa dycKuB"
  >
    <section
      className="sc-bwzfXH fiRAvd"
    >
      <h2
        className="sc-htpNat gKOkZF"
      >
        Search Archive
      </h2>
      <button
        className="MuiButtonBase-root MuiButton-root Footer-howToSearchButton-1 MuiButton-text"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          How to search
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </section>
    <section
      className="sc-bxivhb jbVdgm"
    >
      <div
        className="MuiFormControl-root Footer-formControl-2"
      >
        <div
          aria-label="category"
          className="MuiInputBase-root WithStyles(ForwardRef(InputBase))-root-44 MuiInputBase-formControl Footer-select-7"
          onClick={[Function]}
        >
          <div
            className="MuiSelect-root"
          >
            <div
              aria-haspopup="true"
              aria-pressed="false"
              className="MuiSelect-select MuiSelect-selectMenu MuiInputBase-input WithStyles(ForwardRef(InputBase))-input-45"
              id="select-category"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                className="sc-bZQynM kihKbR"
              >
                Category
              </span>
            </div>
            <input
              id="category-customized-select"
              name="category"
              type="hidden"
              value="Category"
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
        <div
          aria-label="location"
          className="MuiInputBase-root WithStyles(ForwardRef(InputBase))-root-44 MuiInputBase-formControl Footer-select-7"
          onClick={[Function]}
        >
          <div
            className="MuiSelect-root"
          >
            <div
              aria-haspopup="true"
              aria-pressed="false"
              className="MuiSelect-select MuiSelect-selectMenu MuiInputBase-input WithStyles(ForwardRef(InputBase))-input-45"
              id="select-location"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                className="sc-bZQynM kihKbR"
              >
                Location
              </span>
            </div>
            <input
              id="location-customized-select"
              name="location"
              type="hidden"
              value="Location"
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
        <div
          aria-label="year"
          className="MuiInputBase-root WithStyles(ForwardRef(InputBase))-root-44 MuiInputBase-formControl Footer-select-7"
          onClick={[Function]}
        >
          <div
            className="MuiSelect-root"
          >
            <div
              aria-haspopup="true"
              aria-pressed="false"
              className="MuiSelect-select MuiSelect-selectMenu MuiInputBase-input WithStyles(ForwardRef(InputBase))-input-45"
              id="select-year"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                className="sc-bZQynM kihKbR"
              >
                Year
              </span>
            </div>
            <input
              id="year-customized-select"
              name="year"
              type="hidden"
              value="Year"
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        className="MuiFormControl-root Footer-multipleSelectFormControl-3"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled Footer-MultipleSelectInputLabel-6"
          data-shrink={false}
          htmlFor="selectMultipleCheckbox"
        >
          Tags
        </label>
        <div
          aria-label="tags"
          className="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <div
            className="MuiSelect-root"
          >
            <div
              aria-haspopup="true"
              aria-pressed="false"
              className="MuiSelect-select MuiSelect-selectMenu MuiSelect-filled MuiInputBase-input MuiInput-input"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </div>
            <input
              id="selectMultipleCheckbox"
              type="hidden"
              value=""
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
    </section>
    <section
      className="sc-ifAKCX ezvIgM"
    >
      <div
        className="MuiFormControl-root Footer-searchFormControl-8"
      >
        <div
          className="MuiFormControl-root MuiTextField-root Footer-searchTextField-9"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Footer-searchTextFieldLabel-10"
            data-shrink={false}
            htmlFor="outlined-search"
          >
            Search...
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              className="MuiInputBase-input MuiInput-input MuiInputBase-inputTypeSearch MuiInput-inputTypeSearch"
              disabled={false}
              id="outlined-search"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="search"
              value=""
            />
          </div>
        </div>
      </div>
      <section
        className="sc-EHOje khXZcR"
      >
        <button
          className="MuiButtonBase-root MuiButton-root Footer-searchButton-11 MuiButton-text"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Search
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root Footer-resetButton-12 MuiButton-text"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Reset
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </section>
    </section>
  </footer>,
  " ",
]
`;
